src/Components/Endpoints/src/Builder/ResourceCollectionUrlEndpoint.cs

```diff
@@ -0,0 +1,291 @@
+            context.Response.Headers[HeaderNames.Vary] = HeaderNames.AcceptEncoding;
+            context.Response.Headers.ETag = new StringValues(_gzipContentETags);
+        }
+
+        private void WriteNoEncodingHeaders(HttpContext context)
+        {
+            context.Response.Headers.ETag = new StringValues(_contentETag);
+        }
+
+        private static void WriteFingerprintHeaders(HttpContext context)
+        {
+            context.Response.Headers[HeaderNames.CacheControl] = "max-age=31536000, immutable";
+        }
+
+        private static void WriteNonFingerprintedHeaders(HttpContext context)
+        {
+            context.Response.Headers[HeaderNames.CacheControl] = "no-cache, must-revalidate";
+        }
+
+        private static void WriteCommonHeaders(HttpContext context, byte[] contents)
+        {
+            context.Response.ContentType = "application/javascript";
+            context.Response.ContentLength = contents.Length;
+        }
+
+        internal IEnumerable<RouteEndpointBuilder> CreateEndpoints(
+            string fingerprintedResourceCollectionUrl,
+            string fingerprintedGzResourceCollectionUrl,
+            string resourceCollectionUrl,
+            string gzResourceCollectionUrl)
+        {
+            var quality = 1 / (1 + _gzipContent.Length);
+            var encodingMetadata = new ContentEncodingMetadata("gzip", quality);
+
+            var fingerprintedGzBuilder = new RouteEndpointBuilder(
+                FingerprintedGzipContent,
+                RoutePatternFactory.Parse(fingerprintedGzResourceCollectionUrl),
+                -100);
+
+            var fingerprintedBuilder = new RouteEndpointBuilder(
+                FingerprintedContent,
+                RoutePatternFactory.Parse(fingerprintedResourceCollectionUrl),
+                -100);
+
+            var fingerprintedBuilderConeg = new RouteEndpointBuilder(
+                FingerprintedGzipContent,
+                RoutePatternFactory.Parse(fingerprintedResourceCollectionUrl),
+                -100);
+
+            fingerprintedBuilderConeg.Metadata.Add(encodingMetadata);
+
+            var gzBuilder = new RouteEndpointBuilder(
+                GzipContent,
+                RoutePatternFactory.Parse(gzResourceCollectionUrl),
+                -100);
+
+            var builder = new RouteEndpointBuilder(
+                Content,
+                RoutePatternFactory.Parse(resourceCollectionUrl),
+                -100);
+
+            var builderConeg = new RouteEndpointBuilder(
+                Content,
+                RoutePatternFactory.Parse(resourceCollectionUrl),
+                -100);
+
+            builderConeg.Metadata.Add(encodingMetadata);
+
+            yield return fingerprintedGzBuilder;
+            yield return fingerprintedBuilder;
+            yield return fingerprintedBuilderConeg;
+            yield return gzBuilder;
+            yield return builderConeg;
+            yield return builder;
+        }
```

Comment:
Why this and not:
```suggestion
            return [
                fingerprintedGzBuilder,
                fingerprintedBuilder,
                fingerprintedBuilderConeg,
                gzBuilder,
                builderConeg,
                builder,
            ];
        }
```
?

The former does generate quite a bit more code.